### ISSUE TITLE
Let's get this back upto date

### ISSUE THREAD

**Issue by Seebiscuit** [simonblee/generator-maryo#22]
Hey @simonblee. I'm a fan of your generator. I recently used it for a project and the core is solid.

Of course, the last PR was made over a year ago and a lot's changed since. Everything from new view naming conventions (`Marionette.Layout` → `Marionette.LayoutView`) to required view properties (`itemView `→ `childView `in `CollectionView`s). And of course version changes: Mn: 2.4.2, Bb: 1.2.3, _: 1.8.3 (LoDash even more drastic)

Also, I think we need to offer more build tool options: Gulp or none (for npm script support)

Finally, there a few api changes Yeoman pushed out (e.g., Yo deprecated `this.mkdir` and exposed and  instance of [`mem-fs-editor`](https://github.com/sboudrias/mem-fs-editor) onto `this.fs`, [from the [docs](http://yeoman.io/authoring/file-system.html)]).

In short, you did a lot here and it's _really_ good. I want to keep using your generator, but it needs to be updated. What are your thoughts about partnering up and maintaining it (maybe even make a fork for v3.0)?

I'm also thinking of reaching out to some Mn members to help.

**Comment by simonblee** [comment]
Hi @Seebiscuit,

Thanks, I'm glad you've found it useful. Yeah, it's looking a bit old now. The reason it's been left so long is I've been working less and less on Marionette projects so as is often the case, this got neglected. I had a bunch of things I wanted to add but never got around to it. These included Browserify support, wider choice of templates and a customisable directory structure for your app.

There is a more complete generator available [here](https://github.com/mrichard/generator-marionette) which you may be aware of. A while ago, the guys from Marionette reached out to expand mine to be more generic as the other one provided a full server stack which they weren't too keen on. I was happy to participate but nothing eventuated. I've seen a few more pop up as well, one of them [here](https://github.com/Accenture/generator-mario).

Time wise, I'm quite limited. I've got a couple of things on the side and have just started a new job so will be moving country shortly as well. Let's get a list of tasks together to see what sort of time is required to get this up to scratch.

**Comment by Seebiscuit** [comment]
Sorry about the monstrous delay in getting back to you. I've also been up to the gills. 

I was vaguely familiar with `generator-marionette` and visited again just a moment ago. I hate it.

Not only is it rigidly opinionated but it also totally misses the point. A Marionette generator, IMHO, should not define the entire stack. Marionette is a front-end technology; they cut out a huge slice of potential users by offering a Node back-end. Plus, I don't use Handlebars, just straight Underscore, and I really like how `generator-maryo` respects that.

I also tried `generator-mario`. I didn't like it either. Too opinionated, bloated. IMHO, Gulp/Grunt are not the only way to go. I think there needs to be a `npm scripts` option as well. 

Maryo I think strikes the perfect balance. So, what do you think? Let's make `generator-maryo` again.
